### PR TITLE
Fix wheels release

### PR DESCRIPTION
**Pull request by megaserg**
Use latest `cibuildwheel` and fix all errors.

### CODE REVIEW

**Comment by megaserg** [comment]
@doublethefish this builds all the wheels successfully on my fork :) would appreciate if you could take a look!

File: .github/workflows/wheels.yml [hunk 1]
```diff
@@ -35,21 +37,14 @@ jobs:
...
-      # Used to host cibuildwheel
-      - uses: actions/setup-python@v3
-
-      - name: Install cibuildwheel
-        run: python -m pip install cibuildwheel==2.11.2
+      - uses: actions/checkout@v4
 
       - name: Build wheels
-        uses: pypa/cibuildwheel@v2.12.0
+        uses: pypa/cibuildwheel@v2.22.0
```
Comment (doublethefish) [inline .github/workflows/wheels.yml:43]: F.A.B. changes. This upgrade and simplification of the action probably makes all the difference, great stuff

File: src/_tmap/cereal/types/tuple.hpp [hunk 2]
```diff
@@ -95,7 +95,7 @@ namespace cereal
       template <class Archive, class ... Types> inline
       static void apply( Archive & ar, std::tuple<Types...> & tuple )
       {
-        serialize<Height - 1>::template apply( ar, tuple );
+        serialize<Height - 1>::template apply<>( ar, tuple );
```
Comment (doublethefish) [inline src/_tmap/cereal/types/tuple.hpp:98]: It is always better to put more information that "Appease clang" for changes like this.

In future I would like to see the specific "appeasement" that is occurring but in this case I assume it is a warning about template ambiguity.

**Comment by megaserg** [comment]
Thanks @doublethefish !

Sorry, I wasn't very diligent with the commit messages and history, I was expecting that you'd squash the whole PR 😌 

Regarding `clang` change: it is to fix this error:
```
        /Users/runner/work/tmap/tmap/src/_tmap/cereal/types/tuple.hpp:119:85: error: a template argument list is expected after a name prefixed by the template keyword [-Wmissing-template-arg-list-after-template-kw]
          119 |     tuple_detail::serialize<std::tuple_size<std::tuple<Types...>>::value>::template apply( ar, tuple );
              |       
```
which is a relatively recent addition to clang: https://github.com/llvm/llvm-project/pull/80801

Regarding 32 mins: that number is end-to-end but depends on available parallelism (in GitHub free tier?) at the moment. Each job takes about 5 mins (10 mins for macos x86_64) so best case you'd see 10 mins end-to-end, worst case ~2 hours (if everything is sequential).

Thanks again! I believe some of the long-standing issues can be closed now :)

**Comment by doublethefish** [comment]
> Sorry, I wasn't very diligent with the commit messages and history, I was expecting that you'd squash the whole PR 😌

No worries, thanks for the PR. We are very grateful.

> Regarding clang change: it is to fix this error:

Thanks for that it's useful to the specific warning, and edifying to know that I guessed right about the type of warning it was 🤓. As you probably know, `-Wmissing-template-arg-list-after-template-kw` is a warning that only errors because we have warning-as-errors on for that bit of code. Similar template-specific warnings have been around for a while, and problematic; for instance, at one point fixing this exact issue in one compiler, but leaving spaces between the angle-brackets, on another compiler would cause `less-than symbol` errors 🤦🏻‍♂️.

**Comment by megaserg** [comment]
Sounds great @doublethefish ! Would you be willing to release the wheels to PyPI?

**Comment by doublethefish** [comment]
Yes, of course. But it will have to be sometime over the next few days.

**Comment by megaserg** [comment]
Happy New Year @doublethefish! Would now be a good time to release the wheels to PyPI? :)

**Pull request merged**